gc/scheduler/scheduler.go

```diff
@@ -0,0 +1,318 @@
+package scheduler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/metadata"
+	"github.com/containerd/containerd/plugin"
+)
+
+// Config configures the garbage collection policies.
+type Config struct {
```

Comment:
Times should all use `time.Duration`.

Reply:
Toml doesn't support `time.Duration`, all the values would end up being nanos instead

Reply:
I could make this a string type so the interface used in the configuration uses the duration suffixes. I don't really want to declare 2 config types at this point since the scheduler is already is already using time.Duration.

Reply:
https://github.com/BurntSushi/toml#using-the-encodingtextunmarshaler-interface

You can use a type that wraps `time.Duration` which is a common way of handling this for any config format (json, yaml, toml).

Reply:
I will just declare 2 config types and use a marshaler. This configuration object is important since it defines a policy.